File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string myString, string pat) {
    string tmp = "";
    for(char t : myString) {
        if(t == 'A') {
            tmp += 'B';
        }
        else {
            tmp += 'A';
        }
    }
    
    if(tmp.find(pat) != string::npos)
        return 1;
    return 0;
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
    public int solution(String myString, String pat) {
        int answer = 0;
        String reverse = "";
        for(char a : pat.toCharArray()) {
            if(a == 'A') {
                reverse += "B";
            }
            else {
                reverse += "A";
            }
        }
        
        if(myString.indexOf(reverse) != -1) {
            answer = 1;
        }
        return answer;
    }
}